samples/shell/shell_test.py

```diff
@@ -0,0 +1,41 @@
+import time
+import re
+import pytest
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def is_regex_found(regex, connection):
+    """ A helper function telling if a given regex was found in an output"""
+    pattern = re.compile(regex)
+    matched = False
+    timeout = 15  # [sec]
+    time_end = time.time() + timeout
+    while time.time() < time_end:
+        line = connection.readline().decode('UTF-8').strip()
```

Comment:
This was my first approach but it seems there is some issue with `iter_stdout` since it doesn't work reliably with qemu. I agree using it could be better since `handler.log` is connected with `iter_stdout`.